src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -314,6 +314,10 @@ class TransportManagementService implements Serializable {
             script.error message
         }
         
+        if (! gotResponse) {
```

Comment:
Please first print the message `echo("${action} successful.")` and then check for `gotResponse`. In that case, if `gotResponse` is false, then we still will have an information in the logs that the operation finished successfully, but response was not able to be obtained.

Reply:
Artem, I have an unsure point, will there be a case that status code is correct but no response body file? If yes, in this case, should we still throw an error when we found the response body file does not exist or just go ahead because of the expected status code.

Reply:
> will there be a case that status code is correct but no response body file?

I do not know, it might be potentially. And that was the former implementation from Marcus.


> If yes, in this case, should we still throw an error when we found the response body file does not exist or just go ahead

I think, the former implementation was that we print success (meaning the code is as expected) and throw an error that response body cannot be obtained. It looks, very fair to me.